src/main/java/net/coderbot/iris/mixin/MixinTranslationStorage.java

```diff
@@ -0,0 +1,54 @@
+package net.coderbot.iris.mixin;
+
+import net.coderbot.iris.Iris;
+import net.minecraft.client.resource.language.LanguageDefinition;
+import net.minecraft.client.resource.language.TranslationStorage;
+import net.minecraft.resource.ResourceManager;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Mixin(TranslationStorage.class)
+public class MixinTranslationStorage {
+
+    //this is needed to keep track of which language code we need to grab our lang files from
+    private static List<String> languageCodes = new ArrayList<>();
+
+    private static final String LOAD = "load(Lnet/minecraft/resource/ResourceManager;Ljava/util/List;)Lnet/minecraft/client/resource/language/TranslationStorage;";
+
+    @Shadow
+    @Final
+    private Map<String, String> translations;
+
+
+    @Inject(method = "get", at = @At("HEAD"), cancellable = true)
+    private void iris$addLanguageEntries(String key, CallbackInfoReturnable<String> cir) {
+        //minecraft loads us language code by default, and any other code will be right after it
+        //so we also check if the user is loading a special language, and if the shaderpack has support for that language
+        //if they do, we load that, but if they do not, we load "en_us" instead
+        Map<String, Map<String, String>> languageMap = Iris.getCurrentPack().getLangMap();
+        if (!translations.containsKey(key)) {
+            languageCodes.forEach(code -> {
+                if (languageMap.get(code) != null) {
+                    if (languageMap.get(code).get(key) != null) {
+                        cir.setReturnValue(languageMap.get(code).get(key));
+                    }
+                }
```

Comment:
This is what I was getting at with my previous comment.
```suggestion
                Map<String, String> translations = languageMap.get(code);

                if (translations != null) {
                    String translation = translations.get(key);

                    if (translation != null) {
                        cir.setReturnValue(translation);
                    }
                }
```